src/Memcached.php

```diff
@@ -2,252 +2,180 @@
-     * @param MemcachedServer[] $servers list of memcached server configurations
-     * @throws InvalidConfigException
-     * @see setSerializer
-     */
-    public function __construct(?SerializerInterface $serializer = null, array $servers = [])
+    public function __construct()
     {
-        parent::__construct($serializer);
-
-        if (empty($servers)) {
-            $servers = [new MemcachedServer('127.0.0.1')];
-        }
-
-        $this->servers = $servers;
-
-        $this->addServers($this->getMemcached(), $this->servers);
+        $this->initCache();
+        parent::__construct();
     }
 
     /**
-     * Add servers to the server pool of the cache specified
+     * Fetches a value from the cache.
      *
-     * @param \Memcached $cache
-     * @param MemcachedServer[] $servers
-     */
-    private function addServers(\Memcached $cache, array $servers): void
-    {
-        $existingServers = [];
-        if ($this->persistentId !== null) {
-            foreach ($cache->getServerList() as $s) {
-                $existingServers[$s['host'] . ':' . $s['port']] = true;
-            }
-        }
-        foreach ($servers as $server) {
-            $serverAddress = $server->getHost() . ':' . $server->getPort();
-            if (empty($existingServers) || !isset($existingServers[$serverAddress])) {
-                $cache->addServer($server->getHost(), $server->getPort(), $server->getWeight());
-            }
-        }
-    }
-
-    /**
-     * Returns the underlying memcached object.
-     * @return \Memcached the memcached object used by this cache component.
-     * @throws InvalidConfigException if memcached extension is not loaded
+     * @param string $key The unique key of this item in the cache.
+     * @param mixed $default Default value to return if the key does not exist.
+     *
+     * @return mixed The value of the item from the cache, or $default in case of cache miss.
+     *
+     * @throws \Psr\SimpleCache\InvalidArgumentException
+     *   MUST be thrown if the $key string is not a legal value.
      */
-    public function getMemcached(): \Memcached
+    public function get($key, $default = null)
     {
-        if ($this->cache === null) {
-            if (!\extension_loaded('memcached')) {
-                throw new InvalidConfigException('MemCached requires PHP memcached extension to be loaded.');
-            }
+        $value = $this->cache->get($key);
 
-            $this->cache = $this->persistentId !== null ? new \Memcached($this->persistentId) : new \Memcached();
-            if ($this->username !== null || $this->password !== null) {
-                $this->cache->setOption(\Memcached::OPT_BINARY_PROTOCOL, true);
-                $this->cache->setSaslAuthData($this->username, $this->password);
-            }
-            if (!empty($this->options)) {
-                $this->cache->setOptions($this->options);
-            }
+        // TODO store error message
```

Comment:
⚠️

Reply:
Do you think it can be useful or there is no need for that?

Reply:
I think there's no need for it.